cfpb/consumerfinance.gov
# Add FAQ schema block to browse & learn pages

The existing implementation of FAQ schema in Wagtail is customized for use on Ask CFPB pages and includes a very limited set of content types appropriate for Ask answers. 

We've recently worked on two projects -- the conversion of the **Repay Student Debt** interactive tool into static pages & the redesign of the **Submit a Complaint** landing page -- that provide new opportunities to apply FAQ schema, but the Ask FAQ module does not meet the needs of either use case due to its content type limitations.

This PR adds two new ways to implement FAQ schema:
  - a **FAQGroup block** with more content options for Repay (and other learn / browse pages)
  - a **FAQ toggle on the Expandable Group module** for Submit a Complaint (and other pages with expandable FAQs, like [the FOIA FAQ page](https://www.consumerfinance.gov/foia-requests/faqs/)).

While it would be possible to manage both use cases with a single FAQ block, it seems preferable to handle them separately for a couple of reasons, including:
- **Content type conflicts**: To work for the Repay / other potential use cases, the FAQGroup module needs to allow content types (wells, info units, etc) that we don't permit in expandables. 
- **Expandable FAQ content strategy**: We often use expandables for FAQ-like information. It would be helpful to be able to easily add FAQ schema to existing expandables on the site without reflowing their content into a new block, and it might also be useful to consider whether FAQ schema is applicable any time we enter content in an expandable.

## Notes
FAQ schema require a wrapper `FAQPage` designation, but this can only be declared once on a page. Since these new FAQ options make it possible to add multiple blocks with FAQ markup, the `FAQPage` declaration can't be made in the FAQ block templates. I'm currently detecting whether the content section contains a FAQ block at the page level and, based on this, outputting the FAQPage markup on the content section's wrapper div. I don't know if this is the best way to handle it, so I'm open to any alternate approaches... 

## Additions

- `FAQGroup` block
- `faq-group` jinja template
- `FAQGroup` on browse and learn pages
- `FAQ` toggle on ExpandableGroup
- FAQ schema handling in expandable and expandable group templates
- FAQ methods on base page type
- Tests for is_faq_page context setting
- Conditional FAQ schema application on the base page template

## How to test this PR

1. Check out branch
2. Run migrations
3. Verify FAQGroup block:
   - In admin, create a new browse page and add a FAQ block
   - Verify that the content is output as expected
   - Verify the schema implementation by copying the source code into the [schema.org validator](https://validator.schema.org/)
4. Verify FAQ ExpandableGroup option:
   - Navigate to one of the existing FAQ pages in the Wagtail admin and check the FAQ box on one of its expandable groups. 
   - Verify that the content is output as expected
   - Verify the schema implementation by copying the source code into the [schema.org validator](https://validator.schema.org/)


## Screenshots
### FAQ Group block
![FAQ](https://user-images.githubusercontent.com/778171/165122107-cc1223f8-9785-45d0-aa3a-8dfe1e58efbf.png)


### Expandable Group option
<img width="1023" alt="Screen Shot 2022-04-25 at 8 24 26 AM" src="https://user-images.githubusercontent.com/778171/165122132-b1419d4a-028b-479d-ac7f-f0d3af8b6688.png">



## Checklist

  - PR titles are used to generate the change log in [releases](../../releases); good ones make that easier to scan.
  - Consider prefixing, e.g., "Mega Menu: fix layout bug", or "Docs: Update Docker installation instructions".
  - [This repo’s docs](https://cfpb.github.io/consumerfinance.gov/) (edit the files in the `/docs` folder) – for basic, close-to-the-code docs on working with this repo
  - CFGOV/platform wiki on GHE – for internal CFPB developer guidance
  - CFPB/hubcap wiki on GHE – for internal CFPB design and content guidance
